packages/extension-webgazer/src/index.ts

```diff
@@ -0,0 +1,290 @@
+import { JsPsych, JsPsychExtension, JsPsychExtensionParameters } from "jspsych";
+
+// we have to add webgazer to the global window object because webgazer attaches itself to
+// the window when it loads
+declare global {
+  interface Window { webgazer: any; }
+}
+
+interface InitializeParameters extends JsPsychExtensionParameters {
+  round_predictions: boolean;
+  auto_initialize: boolean;
+  sampling_interval: number;
+  webgazer: any;
+}
+
+interface OnStartParameters extends JsPsychExtensionParameters {
+  targets: Array<string>;
+}
+
+class WebGazerExtension implements JsPsychExtension {
+
+  constructor(private jsPsych: JsPsych) { }
+
+  // private state for the extension
+  // extension authors can define public functions to interact
+  // with the state. recommend not exposing state directly
+  // so that state manipulations are checked.
+  private currentTrialData = [];
+  private currentTrialTargets = {};
+  private currentTrialSelectors:Array<string>;
+  private domObserver:MutationObserver;
+  private webgazer;
+  private initialized:boolean = false;
+  private currentTrialStart:number;
+  private activeTrial:boolean = false;
+  private sampling_interval:number;
+  private round_predictions:boolean;
+  private gazeInterval:number;
+  private gazeUpdateCallbacks:Array<any>;
+  private currentGaze:Object;
+
+  // required, will be called at jsPsych.init
+  // should return a Promise
+  initialize({ round_predictions = true, auto_initialize = false, sampling_interval = 34, webgazer }: InitializeParameters): Promise<void> {
```

Comment:
Can I provide jsdoc hints for these parameters that are nested in the params object?

Reply:
Oooh, maybe I do that by applying them to the Type declaration?

Reply:
Exactly!